rest/tests/test_rest.py

```diff
@@ -0,0 +1,579 @@
+        """
+        Test for Vocabulary
+        """
+
+        vocabularies = self.get_vocabularies(self.repo.slug)
+        self.assertEqual(0, vocabularies['count'])
+
+        self.create_vocabulary(self.repo.slug)
+
+        vocabularies = self.get_vocabularies(self.repo.slug)
+        self.assertEqual(1, vocabularies['count'])
+        new_vocab_slug = vocabularies['results'][0]['slug']
+
+        self.get_vocabulary(self.repo.slug, new_vocab_slug)
+
+        # test PUT and PATCH
+        input_dict = {
+            'name': 'other name',
+            'description': 'description',
+            'required': True,
+            'weight': 1000,
+            'vocabulary_type': 'f',
+        }
+
+        # patch with missing slug
+        self.patch_vocabulary(
+            self.repo.slug, "missing", {'name': 'rename'},
+            expected_status=HTTP_404_NOT_FOUND
+        )
+
+        output_dict = self.patch_vocabulary(
+            self.repo.slug, new_vocab_slug, {'name': 'rename'})
+        new_vocab_slug = output_dict['slug']
+        self.assertEqual('rename',
+                         Vocabulary.objects.get(slug=new_vocab_slug).name)
+
+        # put with missing slug
+        self.put_vocabulary(self.repo.slug, "missing slug", input_dict,
+                            expected_status=HTTP_404_NOT_FOUND)
+
+        output_dict = self.put_vocabulary(
+            self.repo.slug, new_vocab_slug, input_dict)
+        new_vocab_slug = output_dict['slug']
+        self.assertEqual(input_dict['name'],
+                         Vocabulary.objects.get(slug=new_vocab_slug).name)
+        # never created a duplicate
+        self.assertEqual(1, Vocabulary.objects.count())
+
+        # try deleting a vocabulary
+        self.delete_vocabulary(self.repo.slug, new_vocab_slug,
+                               expected_status=HTTP_405_METHOD_NOT_ALLOWED)
+
+        vocabularies = self.get_vocabularies(self.repo.slug)
+        self.assertEqual(1, vocabularies['count'])
+
+        # test missing repository
+        self.create_vocabulary("missing", input_dict,
+                               expected_status=HTTP_404_NOT_FOUND)
+        self.get_vocabularies("missing", expected_status=HTTP_404_NOT_FOUND)
+        self.get_vocabulary("missing", "missing",
+                            expected_status=HTTP_404_NOT_FOUND)
+
+        # as anonymous
+        self.logout()
+        self.get_vocabularies("missing", expected_status=HTTP_403_FORBIDDEN)
+        self.get_vocabulary("missing", "missing",
+                            expected_status=HTTP_403_FORBIDDEN)
+
+    def test_other_vocabulary(self):
+        """Test that two repositories have two different vocabularies"""
+
+        # missing fields
+        input_dict = {
+            'name': 'name',
+            'description': 'description',
+        }
+        self.create_vocabulary(self.repo.slug, input_dict,
+                               expected_status=HTTP_400_BAD_REQUEST)
+
+        input_dict = {
```

Comment:
Nit: This looks identical to the `input_dict` above; could have been reused. I know one is used to try to create a vocabulary and the other a repository.